packages/discovery-provider/src/tasks/index_user_bank.py

```diff
@@ -347,11 +347,16 @@ def validate_purchase(
         is_valid = False
     # Check that the recipients all got the correct split
     for account, split in purchase_metadata["splits"].items():
+        if account not in balance_changes:
+            logger.error(
+                f"index_payment_router.py | No split given to account={account}, expected={split}"
+            )
+            return False
```

Comment:
nit: If we're doing early returns, let's get rid of `is_valid` and go all in on it? Or go back to using `is_valid` here? Either way but probably not both

Reply:
It's funny, multiple people made the exact same comment on the payment router indexing PR 🤦 .
You're probably right and we could also probably ditch the first check above about the sender paying full price as our real test condition is "did all the recipients get their full split"

Reply:
Updated to remove the first check and the temp boolean